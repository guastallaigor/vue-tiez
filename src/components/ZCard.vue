<template>
  <div class="z-card" :style="width ? `width:${width}` : ''" :class="{ dark }">
    <slot/>
  </div>
</template>

<script>
export default {
  name: 'z-card',
  props: {
    width: {
      type: String,
      default: 'auto'
    },
    dark: {
      type: Boolean,
      default: false
    }
  },
  data: () => ({})
}
</script>
