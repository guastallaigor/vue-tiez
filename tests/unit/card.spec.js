import { shallowMount } from '@vue/test-utils'
import ZCard from '../../src/components/ZCard'

describe('When I create the ZCard component', () => {
  const createComponent = (propsData = {}, slot) => {
    return shallowMount(ZCard, {
      propsData,
      slots: {
        default: slot || 'ZCard'
      }
    })
  }

  it('should be a Vue instance', () => {
    const wrapper = createComponent()
    expect(wrapper.find(ZCard).isVueInstance()).toBe(true)
  })

  it('should have a width auto and dark false by default', () => {
    const wrapper = createComponent()
    const card = wrapper.find('.z-card')
    expect(card.exists()).toBe(true)
    expect(card.classes().length).toBe(1)
    expect(card.classes()).toEqual(['z-card'])
    expect(card.attributes().style).toBe('width: auto;')
  })

  it('should change the width to 300px when the width prop is passed', () => {
    const wrapper = createComponent({ width: '300px' })
    const card = wrapper.find('.z-card')
    expect(card.exists()).toBe(true)
    expect(card.classes().length).toBe(1)
    expect(card.attributes().style).toBe('width: 300px;')
  })

  it('should add a dark color when dark prop is passed', () => {
    const wrapper = createComponent({ dark: true })
    const card = wrapper.find('.z-card')
    expect(card.exists()).toBe(true)
    expect(card.classes().length).toBe(2)
    expect(card.classes()).toEqual(['z-card', 'dark'])
  })

  it('should match snapshot', () => {
    const wrapper = createComponent({
      width: 'auto',
      dark: false
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
