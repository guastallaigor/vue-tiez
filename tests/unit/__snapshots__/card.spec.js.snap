// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I create the ZCard component should match snapshot 1`] = `<div class="z-card" style="width: auto;">ZCard</div>`;
